sp/src/game/server/hl2/npc_citizen17.cpp

```diff
@@ -1443,29 +1451,45 @@ bool CNPC_Citizen::GiveBackupWeapon( CBaseCombatWeapon * pWeapon, CBaseEntity *
 	// Don't give any more backup weapons!
 	m_bUsedBackupWeapon = true;
 
-	// Is this weapon already a side arm?
-	if ( pWeapon != NULL && pWeapon->ClassMatches( "weapon_smg1" ) || pWeapon->ClassMatches( "weapon_smg2" ) || pWeapon->ClassMatches( "weapon_pistol" ) )
+	if ( pWeapon != NULL )
 	{
-		// Very lucky citizens get crowbars as backup
-		if ( random->RandomInt( 1, 6 ) == 1 )
+		// Is this a melee weapon?
+		if ( pWeapon->IsMeleeWeapon() || pWeapon->ClassMatches( "weapon_crowbar" ) )
 		{
-			CBaseCombatWeapon * pCrowbar = GiveWeaponHolstered( AllocPooledString( "weapon_crowbar" ) );
-			pCrowbar->AddSpawnFlags( SF_WEAPON_NO_PLAYER_PICKUP );
-			pCrowbar->SetName( AllocPooledString( "worthless" ) ); // Bad Cop will say the crowbar pickup line upon interacting with this
-
-			return true;
+			// No more backups
+			return false;
 		}
+		// Is this weapon already a side arm?
+		else if ( pWeapon->ClassMatches( "weapon_smg1" ) || pWeapon->ClassMatches( "weapon_smg2" ) || pWeapon->ClassMatches( "weapon_pistol" ) )
+		{
+			// Very lucky citizens get crowbars as backup
+			if ( RandomInt( 1, 6 ) == 1 )
+			{
+				CBaseCombatWeapon * pCrowbar = GiveWeaponHolstered( AllocPooledString( "weapon_crowbar" ) );
+				pCrowbar->AddSpawnFlags( SF_WEAPON_NO_PLAYER_PICKUP );
```

Comment:
This is not part of your PR but could we remove this line? We made weapon_crowbar nonswappable in the weapon script. Without this line, Bad Cop will be able to pick up (but not equip) the crowbar and will finally say the funny "WORTHLESS" line